grpc-server/workflow.go

```diff
@@ -279,3 +273,17 @@ func (s *server) ShowWorkflowEvents(req *workflow.GetRequest, stream workflow.Wo
 	metrics.CacheHits.With(labels).Inc()
 	return nil
 }
+
+func getWorkflowState(db db.Database, ctx context.Context, id string) workflow.State {
```

Comment:
any chance you could add code comments on why this logic is needed?

Reply:
Yeah sure I will add reason behind this logic as comment over this function.